packages/react/src/tooltip/Tooltip.tsx

```diff
@@ -44,118 +32,29 @@ export const Tooltip = forwardRef<HTMLButtonElement, TooltipProps>(
-
-                  const elements: Element[] = [innerRef.current];
-                  while (!truncated && elements.length) {
-                    const element = elements.shift();
-                    if (!(element instanceof HTMLElement)) {
-                      continue;
-                    }
-                    const { offsetWidth, scrollWidth } = element;
-
-                    if (offsetWidth < scrollWidth) {
-                      truncated = true;
-                      break;
-                    }
-
-                    elements.push(...element.children);
-                  }
-
-                  if (!truncated) {
-                    return;
-                  }
-                }
-
-                setOpen(flag);
-                onOpenChange?.(flag);
-              }
-            : setOpen
-        }
+        keepOpenOnActivation={keepOpenOnActivation}
+        onOpenChange={onOpenChange}
         open={open}
       >
-        <RadixTooltip.Trigger
-          asChild
-          onClick={
-            keepOpenOnActivation
-              ? (event) => {
-                  event.preventDefault();
-                  setOpen(true);
-                }
-              : undefined
-          }
-          ref={ref}
-        >
+        <TooltipTrigger asChild ref={ref}>
           {children}
-        </RadixTooltip.Trigger>
+        </TooltipTrigger>
 
-        <AnimatePresence>
-          {open && (
-            <RadixTooltip.Portal forceMount>
-              <Transition duration="sm" type="pop">
-                <Box
-                  asChild
-                  bg="bg.neutral.inverse"
-                  color="white"
-                  px="12"
-                  py="8"
-                  rounded="md"
-                  z={z}
-                  {...props}
-                >
-                  <RadixTooltip.Content
-                    onPointerDownOutside={
-                      keepOpenOnActivation
-                        ? (event: CustomEvent) => {
-                            if (
-                              innerRef.current?.contains(
-                                event.target as Node | null,
-                              )
-                            )
-                              event.preventDefault();
-                          }
-                        : undefined
-                    }
-                    sideOffset={5}
-                  >
-                    <Text fontSize="sm">{content}</Text>
-                    {withArrow && (
-                      <RadixTooltip.Arrow
-                        fill={theme.colors["neutral.900"]}
-                        height={4}
-                        width={8}
-                      />
-                    )}
-                  </RadixTooltip.Content>
-                </Box>
-              </Transition>
-            </RadixTooltip.Portal>
-          )}
-        </AnimatePresence>
-      </RadixTooltip.Root>
+        <TooltipContent sideOffset={5} z={z} {...props}>
```

Comment:
let's move this sideOffset=5 default to the `TooltipContent` component (which currently has sideOffset=2)

and let's remove the `z` from here
```suggestion
        <TooltipContent {...props}>
```